ftw/linkchecker/report_generating.py

```diff
@@ -98,3 +109,101 @@ def safe_workbook(self):
     def get_workbook(self):
         self.output_xlsx.seek(0)
         return self.output_xlsx
+
+
+class ReportHandler(object):
+
+    def __init__(self, site, accumulator):
+        self._site = site
+        self._accumulator = accumulator
+
+        self._xlsx_file = self._create_excel_report(
+                (self._accumulator.internal_broken_link_objs +
+                    self._accumulator.external_broken_link_objs),
+                self._site.configuration.base_uri)
+        self._xlsx_file_content = self._xlsx_file.read()
+        self._file_name = self._generate_file_name()
+
+    def send_report(self):
+        self._send_mail_with_excel_report_attached(
+            self._site.configuration.email,
+            self._site.obj,
+            self._accumulator.time_external_routine,
+            self._xlsx_file_content,
+            self._file_name)
+
+    def upload_report(self):
+        if self._site.configuration.upload_location:
+            self._upload_report_to_filelistingblock(
+                self._site.configuration.upload_location,
+                self._xlsx_file,
+                self._file_name)
+
+    def _generate_file_name(self):
+        return u'linkchecker_report_{}.xlsx'.format(
+            time.strftime('%Y_%b_%d_%H%M%S', time.gmtime()))
+
+    def _create_excel_report(self, link_objs, base_uri):
+        xlsx_file = self._create_excel_report_and_return_filepath(
+            link_objs, base_uri)
+        return xlsx_file
+
+    def _create_excel_report_and_return_filepath(self, link_objs, base_uri):
```

Comment:
Normally the return parameter is not in the function name

Reply:
I removed that part of the name in 8f1eb8a.